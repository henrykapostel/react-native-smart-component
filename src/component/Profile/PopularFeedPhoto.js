import React, { Component } from 'react'
import { View, Image, TouchableOpacity } from 'react-native'
import { styles } from './style'
import { WINDOW_WIDTH } from './../../constants'

export class PopularFeedPhoto extends Component {
  constructor(props) {
    super(props)
    this.state = {
      popularFeedPhoto: props.data
    }
  }
  
  onPressPopularItem = (index) => {
    const { popularFeedPhoto } = this.state
    alert(popularFeedPhoto.posts[index].objectId)
  }
  
  renderPopularImage = (item, index) => {
    return (
      <TouchableOpacity key={index} onPress={() => this.onPressPopularItem(index)}>
        <Image
          source={{ uri: item.thumbnail }}
          style={[styles.popularImage]}
        />
      </TouchableOpacity>
    )
  }
  
  renderContent() {
    const { popularFeedPhoto } = this.state
    if (popularFeedPhoto.posts) {
      return popularFeedPhoto.posts.map((item, index) => {
        return this.renderPopularImage(item, index)
      })
    }
    return <View />
  }
  
  render() {
    const { popularFeedPhoto } = this.state
    let height = 100
    if (popularFeedPhoto.posts) {
      height = Math.ceil(popularFeedPhoto.posts.length / 3) * (WINDOW_WIDTH / 3)
    }
    return (
      <View style={[styles.popularFeedPhotoContainer, { height }]}>
        {this.renderContent()}
      </View>
    )
  }
}