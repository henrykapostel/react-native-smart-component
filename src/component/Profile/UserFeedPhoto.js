import React, { Component } from 'react'
import { View } from 'react-native'
import Carousel, { Pagination } from 'react-native-snap-carousel'
import { WINDOW_WIDTH, ITEM_WIDTH } from './../../constants'
import { SliderEntry } from './../common'
import { styles } from './style'

export class UserFeedPhoto extends Component {
  
  constructor(props) {
    
    super(props)
    this.state = {
      userFeedPhoto     : props.data,
      sliderRef        : null,
      sliderActiveSlide: 1
    }
  }
  
  onSetRef = (ref) => {
    
    const { sliderRef } = this.state
    if (!sliderRef) {
      this.setState({ sliderRef: ref })
    }
  }
  
  onSnapToItem = (index) => {
    
    this.setState({ sliderActiveSlide: index })
  }
  
  renderItemWithParallax = ({ item }, parallaxProps) => {
    
    return (
      <SliderEntry
        data={item}
        even={false}
        parallax={true}
        parallaxProps={parallaxProps}
      />
    )
  }
  
  renderContent() {
    const { userFeedPhoto } = this.state
    
    if (userFeedPhoto.posts) {
      return (
        <Carousel
          ref={this.onSetRef}
          data={userFeedPhoto.posts}
          renderItem={this.renderItemWithParallax}
          sliderWidth={WINDOW_WIDTH}
          itemWidth={ITEM_WIDTH}
          hasParallaxImages={true}
          firstItem={1}
          inactiveSlideScale={0.94}
          inactiveSlideOpacity={0.7}
          enableMomentum={false}
          containerCustomStyle={styles.slider}
          contentContainerCustomStyle={styles.sliderContentContainer}
          loop={true}
          loopClonesPerSide={3}
          autoplay={false}
          autoplayDelay={500}
          autoplayInterval={3000}
          onSnapToItem={this.onSnapToItem}
        />
      )
    }
    return <View />
  }
  
  renderPagination() {
    const { userFeedPhoto, sliderActiveSlide , sliderRef} = this.state
  
    if (userFeedPhoto.posts) {
      return (
        <Pagination
          dotsLength={userFeedPhoto.posts.length}
          activeDotIndex={sliderActiveSlide}
          containerStyle={styles.paginationContainer}
          dotColor="steelblue"
          dotStyle={styles.paginationDot}
          inactiveDotColor="black"
          inactiveDotOpacity={0.4}
          inactiveDotScale={0.6}
          carouselRef={sliderRef}
          tappableDots={!!sliderRef}
        />
      )
    }
    return <View />
  }
  
  render() {
    return (
      <View style={styles.userFeedPhotoContainer}>
        {this.renderContent()}
        {this.renderPagination()}
      </View>
    )
  }
}