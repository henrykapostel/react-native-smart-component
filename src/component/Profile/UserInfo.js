import React, { Component } from 'react'
import {
  View,
  Image,
  Text,
} from 'react-native'
import Autolink from 'react-native-autolink'
import ReadMore from '@expo/react-native-read-more-text'
import { styles } from './style'

export class UserInfo extends Component {
  constructor(props) {
    super(props)
    this.state = {
      info: props.data
    }
  }
  
  onPressLink = (url, linkObject) => {
    alert(linkObject.matchedText)
  }
  
  onDesLayout = ({nativeEvent}) => {
    console.info('nativeEvent', nativeEvent)
  }
  
  renderTruncatedFooter = (handlePress) => {
    return (
      <Text style={styles.footerText} onPress={handlePress}>
        Read more
      </Text>
    )
  }
  
  renderRevealedFooter = (handlePress) => {
    return (
      <Text style={styles.footerText} onPress={handlePress}>
        Show less
      </Text>
    )
  }
  
  renderDescription() {
    const { info } = this.state
    return (
      <View style={styles.descriptionContainer}>
        {info.bio &&
        <ReadMore
          numberOfLines={2}
          renderTruncatedFooter={this.renderTruncatedFooter}
          renderRevealedFooter={this.renderRevealedFooter}
        >
          <Autolink
            text={info.bio}
            style={styles.descriptionText}
            numberOfLines={3}
            hashtag="instagram"
            mention="twitter"
            onPress={this.onPressLink}
            url={true}
            onLayout={this.onDesLayout}
          />
        </ReadMore>}
      </View>
    )
  }
  
  render() {
    const { info } = this.state
    return (
      <View style={styles.userInfoContainer}>
        <Image source={{uri: info.profileThumbnail }} style={styles.userPhoto}/>
        <View style={styles.userInfoSubContainer}>
          <Text style={styles.nameTitleText}>
            {info.name}
          </Text>
          {this.renderDescription()}
        </View>
      </View>
    )
  }
}