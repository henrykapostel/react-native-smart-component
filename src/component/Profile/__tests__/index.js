import 'react-native'
import React from 'react'
import { Profile } from './..'
import { UserInfo } from './../UserInfo'
import { UserFeedPhoto } from './../UserFeedPhoto'
import { PopularFeedPhoto } from './../PopularFeedPhoto'

import renderer from 'react-test-renderer'

it('Profile renders correctly', () => {
  const tree = renderer.create(
    <Profile userInfo={{}} userFeedPhoto={{}} popularFeedPhoto={{}}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('UserInfo renders correctly', () => {
  const tree = renderer.create(
    <UserInfo data={{}}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('UserFeedPhoto renders correctly', () => {
  const tree = renderer.create(
    <UserFeedPhoto data={{}}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('PopularFeedPhoto renders correctly', () => {
  const tree = renderer.create(
    <PopularFeedPhoto data={{}}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})