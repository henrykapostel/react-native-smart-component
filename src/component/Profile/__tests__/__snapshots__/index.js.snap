// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularFeedPhoto renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexWrap": "wrap",
        "height": 450,
      },
      Object {
        "height": 100,
      },
    ]
  }
>
  <View />
</View>
`;

exports[`Profile renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingTop": 10,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 45,
            "height": 90,
            "width": 90,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        />
        <View
          style={
            Object {
              "marginTop": 10,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "height": 583.6,
        }
      }
    >
      <View />
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "flexWrap": "wrap",
            "height": 450,
          },
          Object {
            "height": 100,
          },
        ]
      }
    >
      <View />
    </View>
  </View>
</RCTScrollView>
`;

exports[`UserFeedPhoto renders correctly 1`] = `
<View
  style={
    Object {
      "height": 583.6,
    }
  }
>
  <View />
  <View />
</View>
`;

exports[`UserInfo renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "black",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 45,
        "height": 90,
        "width": 90,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    />
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    />
  </View>
</View>
`;
